src/channel_manager.ts

```diff
@@ -0,0 +1,527 @@
+    const { pagination, channels, ready } = this.state.getLatestValue();
+    const { filters, sort, options, isLoadingNext, hasNext } = pagination;
+
+    if (!ready || isLoadingNext || !hasNext) {
+      return;
+    }
+
+    try {
+      const { offset = 0, limit = 10 } = options;
+      this.state.partialNext({
+        pagination: { ...pagination, isLoading: false, isLoadingNext: true },
+      });
+      const nextChannels = await this.client.queryChannels(filters, sort, options, this.stateOptions);
+      const newOffset = offset + (nextChannels?.length ?? 0);
+      const newOptions = { ...options, offset: newOffset };
+
+      this.state.partialNext({
+        channels: [...(channels || []), ...nextChannels],
+        pagination: {
+          ...pagination,
+          hasNext: (nextChannels?.length ?? 0) >= limit,
+          isLoading: false,
+          isLoadingNext: false,
+          options: newOptions,
+        },
+      });
+    } catch (error) {
+      this.client.logger('error', (error as Error).message);
+      this.state.next((currentState) => ({
+        ...currentState,
+        pagination: { ...currentState.pagination, isLoadingNext: false },
+      }));
+      throw error;
+    }
+  };
+
+  private notificationAddedToChannelHandler = async (event: Event<SCG>) => {
+    const { id, type, members } = event?.channel ?? {};
+    if (type && this.options.allowNewMessagesFromUnfilteredChannels) {
+      const channel = await getAndWatchChannel({
+        client: this.client,
+        id,
+        members: members?.reduce<string[]>((acc, { user, user_id }) => {
+          const userId = user_id || user?.id;
+          if (userId) {
+            acc.push(userId);
+          }
+          return acc;
+        }, []),
+        type,
+      });
+      const { pagination, channels } = this.state.getLatestValue();
+      if (!channels) {
+        return;
+      }
+
+      const { sort } = pagination ?? {};
+
+      this.setChannels(
+        moveChannelUpwards({
+          channels,
+          channelToMove: channel,
+          sort,
+        }),
+      );
+    }
+  };
+
+  private channelDeletedHandler = (event: Event<SCG>) => {
+    const { channels } = this.state.getLatestValue();
+    if (!channels) {
+      return;
+    }
+
+    const newChannels = [...channels];
+    const channelIndex = newChannels.findIndex((channel) => channel.cid === (event.cid || event.channel?.cid));
+
+    if (channelIndex < 0) {
+      return;
+    }
+
+    newChannels.splice(channelIndex, 1);
+    this.setChannels(newChannels);
+  };
+
+  private channelHiddenHandler = this.channelDeletedHandler;
+
+  private newMessageHandler = (event: Event<SCG>) => {
+    const { pagination, channels } = this.state.getLatestValue();
+    if (!channels) {
+      return;
+    }
+    const { filters, sort } = pagination ?? {};
+
+    const channelType = event.channel_type;
+    const channelId = event.channel_id;
+
+    if (channelType && channelId) {
```

Comment:
Can we do this to prevent nesting?:

```suggestion
  if (!channelType || !channelId) return;
```

Reply:
Yep, thanks !